scapy/layers/dns.py

```diff
@@ -437,6 +437,23 @@ def compress(self):
         """Return the compressed DNS packet (using `dns_compress()`"""
         return dns_compress(self)
 
+    def pre_dissect(self, s):
+        """
+        Check that a valid DNS over TCP message can be decoded
+        """
+        if isinstance(self.underlayer, TCP):
+            # Compute the length of the DNS packet
+            dns_len = 0
+            if len(s) >= 2:
+                dns_len = struct.unpack("!H", s[:2])[0]
```

Comment:
Maybe we should raise the ScapyException error here too?

Reply:
You mean to explicitly raise an exception if `len(s) < 2` ?

Reply:
Yep. I think it would make sense.

Reply:
Done.